dashboard-app/src/features/dashboard/components/DataCard/util.ts

```diff
@@ -0,0 +1,20 @@
+
+export const sentenceCreator = (xs: string[]) => {
+  switch (xs.length) {
+    case 0:
+    case 1:
+      return xs;
+    case 2:
+      return [xs[0], 'and', xs[1]].map((x) => x.replace('and', ' and '));
+    case 3:
+    default:
```

Comment:
This case is more complicated than it needs to be. This handles `xs.length > 1`:
```ts
xs.slice(0, -1).join(', ').concat(' and ').concat(xs.slice(-1).join())
```

Reply:
The above example returns a `string` rather than `string[]`. The issue I was having was inserting a comma element after every word. I can do a mash up of the two to remove the ternary

Reply:
Maybe you can look at `intersperse` in ramda

Reply:
```ts
R.intersperse(', ', xs.slice(0, -1)).concat('and').concat(xs.slice(-1))
```